src/cclib/io/wfxwriter.py

```diff
@@ -0,0 +1,475 @@
+        to_bohr = lambda x: utils.convertor(x, 'Angstrom', 'bohr')
+        nuc_coords = [coord_template % tuple(to_bohr(coord))
+                      for coord in self.ccdata.atomcoords[-1]]
+        return nuc_coords
+
+    def _net_charge(self):
+        """Net charge on molecule."""
+        return '%22.11E' % self.ccdata.charge
+
+    def _no_electrons(self):
+        """Number of electrons in molecule."""
+        homos = self.ccdata.homos
+        charge = self.ccdata.charge if hasattr(self.ccdata, 'charge') else 0.0
+        if len(homos) > 1:
+            return int(sum(homos)) + 2
+        else:
+            return int((homos[0] + 1)*2 - (charge % 2))
+
+    def _no_alpha_electrons(self):
+        """Number of alpha electrons in molecule."""
+        return int(numpy.ceil(self._no_electrons() / 2.0))
+
+    def _no_beta_electrons(self):
+        """Number of beta electrons in molecule."""
+        return self._no_electrons() - self._no_alpha_electrons()
+
+    def _spin_mult(self):
+        """Electronic Spin Multiplicity"""
+        return self.ccdata.mult
+
+    def _prim_centers(self):
+        """List of nuclear numbers upon which the primitive basis functions
+        are centered."""
+        prim_centers = []
+        for nuc_num, atom in enumerate(self.ccdata.gbasis, start=1):
+            for prims in atom:
+                prim_centers += [nuc_num] * self.num_orb[prims[0]]\
+                                * len(prims[1])
+
+        return self._list_format(prim_centers, 10, '%d ')
+
+    def _rearrange_modata(self, data):
+        """Rearrange MO related data."""
+        prim_types = self._prim_types()
+        if isinstance(data, numpy.ndarray):
+            data = data.tolist()
+
+        pos_yyx = [key for key, val in enumerate(prim_types)
+                   if val == 17]
+        pos_yyz = [key for key, val in enumerate(prim_types)
+                   if val == 16]
+
+        if pos_yyx:
+            for pos in pos_yyx:
+                data.insert(pos-3, data.pop(pos))
+        if pos_yyz:
+            for pos in pos_yyz:
+                data.insert(pos+3, data.pop(pos + 1))
+
+        return data
+
+
+    def _prim_types(self):
+        """Primitive type numbering starts at 1.
+        The definition of the Cartesian Gaussian primitive types is:
+        1 S, 2 PX, 3 PY, 4 PZ, 5 DXX, 6 DYY, 7 DZZ, 8 DXY, 9 DXZ, 10 DYZ,
+        11 FXXX, 12 FYYY, 13 FZZZ, 14 FXXY, 15 FXXZ, 16 FYYZ, 17 FXYY,
+        18 FXZZ, 19 FYZZ, 20 FXYZ,
+        21 GXXXX, 22 GYYYY, 23 GZZZZ, 24 GXXXY, 25 GXXXZ, 26 GXYYY,
+        27 GYYYZ, 28 GXZZZ,
+        29 GYZZZ, 30 GXXYY, 31 GXXZZ, 32 GYYZZ, 33 GXXYZ, 34 GXYYZ,
+        35 GXYZZ,
+        36 HZZZZZ, 37 HYZZZZ, 38 HYYZZZ, 39 HYYYZZ,
+        40 HYYYYZ, 41 HYYYYY, 42 HXZZZZ, 43 HXYZZZ, 44 HXYYZZ,
+        45 HXYYYZ, 46 HXYYYY, 47 HXXZZZ, 48 HXXYZZ, 49 HXXYYZ,
+        50 HXXYYY, 51 HXXXZZ, 52 HXXXYZ, 53 HXXXYY, 54 HXXXXZ, 55 HXXXXY,
+        56 HXXXXX
+        """
```

Comment:
_*Not a thorough investigation_, 
for `MoOCl4` molecule in spherical basis :
(values from ccget on `Gaussian`)
```
'Mo1_1S', 'Mo1_2S',
'Mo1_3PX', 'Mo1_3PY', 'Mo1_3PZ', 
'Mo1_4PX', 'Mo1_4PY', 'Mo1_4PZ',
'Mo1_5D 0', 'Mo1_5D+1', 'Mo1_5D-1', 'Mo1_5D+2', 'Mo1_5D-2'
```
, the primitives are numbered as:
(values from `.wfx` written for `Gaussian` by AIMALL)
```
1 1
2 3 4 
2 3 4 
5 6 7 8 9 10
```

and in cartesian basis:
(values from ccget on `GAMESS`)
```
'Mo1_S', 'Mo1_S', 'Mo1_S',
'Mo1_X', 'Mo1_Y', 'Mo1_Z', 
'Mo1_X', 'Mo1_Y', 'Mo1_Z', 
'Mo1_X', 'Mo1_Y', 'Mo1_Z', 
'Mo1_XX', 'Mo1_YY', 'Mo1_ZZ', 'Mo1_XY', 'Mo1_XZ', 'Mo1_YZ', 
'Mo1_XX', 'Mo1_YY', 'Mo1_ZZ', 'Mo1_XY', 'Mo1_XZ', 'Mo1_YZ'
```
the primitve types look something like:
(values from `.wfn` written for `GAMESS`)
```
1 1 1 1 1 1 1 1 2 2 
2 3 3 3 4 4 4 2 2 3 
3 4 4 2 3 4 5 5 5 6 
6 6 7 7 7 8 8 8 9 9 
9 10 10 10 5 6 7 8 9 10 
```
The repetitions are due to basis set involved, which are handled using `gbasis` in the writer.